bootstrap/kubeadm/main.go

```diff
@@ -116,38 +115,50 @@ func main() {
 		os.Exit(1)
 	}
 
-	// Kubeadm controllers.
-	if err = (&kubeadmbootstrapcontrollers.KubeadmConfigReconciler{
+	setupWebhooks(mgr)
+	setupReconcilers(mgr)
+
+	// +kubebuilder:scaffold:builder
+	setupLog.Info("starting manager")
+	if err := mgr.Start(ctrl.SetupSignalHandler()); err != nil {
+		setupLog.Error(err, "problem running manager")
+		os.Exit(1)
+	}
+}
+
+func setupReconcilers(mgr ctrl.Manager) {
+	if webhookPort != 0 {
```

Comment:
Instead of triggering on webhookPort, would it make sense to add flags to enable either/or? That way we could leave the default webhookPort alone and have the webhook deployment pass the flag to enable webhooks, and the controller deployment pass the flag to enable reconciliation.